### PR TITLE
task/TUI-198 - Breadcrumbs in Files

### PR DESCRIPTION
**Pull request by nathandf**
## Overview:
Added breadcrumbs to track and navigate folders

## Related Github Issues:
- [TUI-198](https://github.com/tapis-project/tapis-ui/issues/198)

## Summary of Changes:
- Added BreadcrumbType
- Function to parse the path and create BreadcrumbType objects
- Unit tests for breadcrumbFromPathname function

## Testing Steps:
1. Navigate to files
2. Click a system in the SystemsNav
3. Click a folder in the FilesListing
4. Click one of the Links in the breadcrumbs to navigate to that directory

## UI Photos:
https://user-images.githubusercontent.com/59402772/137527496-91b1ab29-41b6-4c43-99ed-2b42d788ff05.mov

### CODE REVIEW

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx [hunk 1]
```diff
@@ -1,19 +1,35 @@
...
+}) => {
   return (
     <div className={styles.root}>
-      Files /
-      {items.map((item) => {
-        return <span> {item} /</span>;
-      })}
+      {breadcrumbs.length === 0
+        ? defaultMessage
+        : breadcrumbs.map((item, index) => {
```
[the remark this answers is not in the document]
Comment (nathandf) [reply src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:24]: Should we add a boolean 'truncates' property that gives control of truncation to the parent component?

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ breadcrumbs }) => {
     let truncatedBreadcrumbs = breadcrumbs.slice(0, 2);
```
Comment (jchuahtacc) [inline src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:17]: suggest:

```
const truncatedBreadcrumbs = breadcrumbs.slice(0, 2)
```

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx [hunk 3]
```diff
@@ -29,7 +29,7 @@ const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ breadcrumbs }) => {
   return (
     <div>
       {breadcrumbs.map((item, index) => {
```
Comment (jchuahtacc) [inline src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:31]: use `truncatedBreadcrumbs` here, and do not reassign value of breadcrumbs param

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx [hunk 4]
```diff
@@ -29,7 +29,7 @@ const Breadcrumbs: React.FC<BreadcrumbsProps> = ({ breadcrumbs }) => {
...
-        return index === breadcrumbs.length - 1 || item.to === undefined ? (
+        return index === breadcrumbs.length - 1 || !item.to ? (
           <span> {item.text} /</span>
```
Comment (jchuahtacc) [inline src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:33]: the spans require keys property. maybe use uuidv4

**Review by jchuahtacc — CHANGES_REQUESTED** [review]

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx (hunk 3 above)
Comment (nathandf) [reply src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:31]: truncatedBreadcrumbs now set to breadcrumbs and not the other way around

File: src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx (hunk 4 above)
Comment (nathandf) [reply src/tapis-ui/_common/Breadcrumbs/Breadcrumbs.tsx:33]: That what I was thinking but hated to go that route. Oh well. uuidv4 it is

**Pull request merged**